Refit/RefitSettings.cs

```diff
@@ -13,21 +12,44 @@ namespace Refit
 {
     public class RefitSettings
     {
+        JsonSerializerSettings jsonSerializerSettings;
+
         public RefitSettings()
         {
             UrlParameterFormatter = new DefaultUrlParameterFormatter();
             FormUrlEncodedParameterFormatter = new DefaultFormUrlEncodedParameterFormatter();
+            ContentSerializer = new JsonContentSerializer();
         }
 
         public Func<Task<string>> AuthorizationHeaderValueGetter { get; set; }
         public Func<HttpMessageHandler> HttpMessageHandlerFactory { get; set; }
 
-        public JsonSerializerSettings JsonSerializerSettings { get; set; }
+        [Obsolete("Set RefitSettings.ContentSerializer = new JsonContentSerializer(JsonSerializerSettings) instead.", false)]
+        public JsonSerializerSettings JsonSerializerSettings
+        {
+            get => jsonSerializerSettings;
+            set
+            {
+                jsonSerializerSettings = value;
+                ContentSerializer = new JsonContentSerializer(value);
+            }
+        }
+
+        public IContentSerializer ContentSerializer { get; set; }
         public IUrlParameterFormatter UrlParameterFormatter { get; set; }
         public IFormUrlEncodedParameterFormatter FormUrlEncodedParameterFormatter { get; set; }
         public bool Buffered { get; set; } = true;
     }
 
+    public interface IContentSerializer
+    {
+        HttpContent Serialize(object item);
+
+        Task<object> DeserializeAsync(HttpContent content, Type objectType);
```

Comment:
Can this be `Task<T> DeserializeAsync<T>(HttpContent content)` to improve type safety?

Reply:
Also, why do we need both a sync & async deserialize? if we need it, do we also need an async serialize?